crates/ty_python_semantic/src/program.rs

```diff
@@ -28,11 +28,15 @@ impl Program {
             search_paths,
         } = settings;
 
-        tracing::info!("Python version: Python {python_version}, platform: {python_platform}");
-
         let search_paths = SearchPaths::from_settings(db, &search_paths)
             .with_context(|| "Invalid search path settings")?;
 
+        let python_version = python_version
+            .or_else(|| search_paths.python_version())
+            .unwrap_or(PythonVersion::latest_ty());
```

Comment:
This is the crux of it, really.

The main implementation question is whether we want to store the `PythonVersion` on `SearchPaths`. It's a little weird, but that's the only place we have a `PythonEnvironment`.

Reply:
Hmm, I don't love it but it seems okay. It's a bit dangerous because it's very easy to take the `python_version` from `ProgramSettings` directly (did you review all read paths?). In fact, you did exactly that in `update_from_settings`, which, I think, is incorrect. 


The reason why I think it's fine is because the logic is encapsulated in here.